core/src/main/java/brooklyn/catalog/internal/CatalogInitialization.java

```diff
@@ -0,0 +1,377 @@
+                new ResourceUtils(this).getResourceAsString(additionsUri), force);
+            
+            if (!hasRunAdditions)
+                log.debug("Added to catalog from CLI: "+items);
+            else
+                log.debug("Added to catalog from CLI: count "+Iterables.size(items));
+            
+            hasRunAdditions = true;
+        }
+    }
+
+    protected void populateViaCallbacks(BasicBrooklynCatalog catalog) {
+        for (Function<CatalogInitialization, Void> callback: callbacks)
+            callback.apply(this);
+    }
+
+    private boolean setFromCatalogLoadMode = false;
+
+    /** @deprecated since introduced in 0.7.0, only for legacy compatibility with 
+     * {@link CatalogLoadMode} {@link BrooklynServerConfig#CATALOG_LOAD_MODE},
+     * allowing control of catalog loading from a brooklyn property */
+    @Deprecated
+    public void applyCatalogLoadMode() {
+        if (setFromCatalogLoadMode) return;
+        setFromCatalogLoadMode = true;
+        Maybe<Object> clmm = ((ManagementContextInternal)managementContext).getConfig().getConfigRaw(BrooklynServerConfig.CATALOG_LOAD_MODE, false);
+        if (clmm.isAbsent()) return;
+        brooklyn.catalog.CatalogLoadMode clm = TypeCoercions.coerce(clmm.get(), brooklyn.catalog.CatalogLoadMode.class);
+        log.warn("Legacy CatalogLoadMode "+clm+" set: applying, but this should be changed to use new CLI --catalogXxx commands");
+        switch (clm) {
+        case LOAD_BROOKLYN_CATALOG_URL:
+            reset = true;
+            break;
+        case LOAD_BROOKLYN_CATALOG_URL_IF_NO_PERSISTED_STATE:
+            // now the default
+            break;
+        case LOAD_PERSISTED_STATE:
+            disallowLocal = true;
+            break;
+        }
+    }
+
+    /** makes the catalog, warning if persistence is on and hasn't run yet 
+     * (as the catalog will be subsequently replaced) */
+    public void populateBestEffort(BasicBrooklynCatalog catalog) {
+        synchronized (mutex) {
+            if (hasRunOfficial || hasRunBestEffort || isPopulating) return;
+            // if a thread calls back in to this, ie calling to it from a getCatalog() call while populating,
+            // it will own the mutex and observe isRunningBestEffort, returning quickly 
+            isPopulating = true;
+            try {
+                if (isStartingUp) {
+                    log.warn("Catalog access requested when not yet initialized; populating best effort rather than through recommended pathway. Catalog data may be replaced subsequently.");
+                }
+                populateCatalog(catalog, true, true, null);
+            } finally {
+                hasRunBestEffort = true;
+                isPopulating = false;
+            }
+        }
+    }
+
+    public void setStartingUp(boolean isStartingUp) {
+        this.isStartingUp = isStartingUp;
+    }
+
+    public void setFailOnStartupErrors(boolean startupFailOnCatalogErrors) {
```

Comment:
Personal preference to have the setters at the top - they are presumably being used like a builder'y type of thing. They are conveniences, rather than having a huge number of args in the constructor. Having all of those together at the top make it easier to see how the behaviour of the class is set.